syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.rsm";
option java_outer_classname = "RsmProto";
option objc_class_prefix = "HLW";

package rsm;

service RSM {
  rpc AppendEntries (AppendEntriesRequest) returns (AppendEntriesResponse) {}
  rpc GetCommitIndex (GetCommitIndexRequest) returns (GetCommitIndexResponse) {}
}

message LogEntry {
  string command = 1;
  string key = 2;
  string value = 3; 
}

message AppendEntriesRequest {
  int64 index = 1;
  repeated LogEntry entries = 2;
}

message AppendEntriesResponse {
  bool success = 1;
  int64 index = 2;
}

message GetCommitIndexRequest {
  
}

message GetCommitIndexResponse {
  int64 commitindex = 1;
}
